src/geometry/web_mercator_rect.rs

```diff
@@ -0,0 +1,140 @@
+        Some(Self {
+            north_west,
+            south_east,
+        })
+    }
+}
+
+/// This is calculating the volume of all points in space which, when projected
+/// to Web Mercator, fall into the given rectangle.
+/// Implemented by extruding the rectangle's four corners along their altitude
+/// axis up and down, which results in a convex polyhedron.
+impl ConvexPolyhedron<f64> for WebMercatorRect {
+    fn compute_corners(&self) -> [Point3<f64>; 8] {
+        let n_w = self.north_west.to_lat_lng();
+        let s_e = self.south_east.to_lat_lng();
+        let ecef_point = |lat: WGS84<f64>, lng: WGS84<f64>, elevation: f64| -> Point3<f64> {
+            let lat_lng = WGS84::new(lat.latitude_degrees(), lng.longitude_degrees(), elevation);
+            let ecef = ECEF::from(lat_lng);
+            Point3::new(ecef.x(), ecef.y(), ecef.z())
+        };
+        [
+            ecef_point(n_w, n_w, MIN_ELEVATION_M), // NW down
+            ecef_point(n_w, s_e, MIN_ELEVATION_M), // NE down
+            ecef_point(s_e, s_e, MIN_ELEVATION_M), // SE down
+            ecef_point(s_e, n_w, MIN_ELEVATION_M), // SW down
+            ecef_point(n_w, n_w, MAX_ELEVATION_M), // NW up
+            ecef_point(n_w, s_e, MAX_ELEVATION_M), // NE up
+            ecef_point(s_e, s_e, MAX_ELEVATION_M), // SE up
+            ecef_point(s_e, n_w, MAX_ELEVATION_M), // SW up
+        ]
+    }
+
+    fn intersector(&self) -> Intersector<f64> {
+        let corners = self.compute_corners();
+        let edges = ArrayVec::from([
+            Unit::new_normalize(corners[1] - corners[0]), // N edge, down
+            Unit::new_normalize(corners[2] - corners[1]), // E edge, down
+            Unit::new_normalize(corners[3] - corners[2]), // S edge, down
+            Unit::new_normalize(corners[0] - corners[3]), // W edge, down
+            Unit::new_normalize(corners[5] - corners[4]), // N edge, up
+            Unit::new_normalize(corners[6] - corners[5]), // E edge, up
+            Unit::new_normalize(corners[7] - corners[6]), // S edge, up
+            Unit::new_normalize(corners[4] - corners[7]), // W edge, up
+            Unit::new_normalize(corners[4] - corners[0]), // NW edge
+            Unit::new_normalize(corners[5] - corners[1]), // NE edge
+            Unit::new_normalize(corners[6] - corners[2]), // SE edge
+            Unit::new_normalize(corners[7] - corners[3]), // SW edge
+        ]);
+
+        let mut face_normals = ArrayVec::new();
+        face_normals.push(Unit::new_normalize(edges[0].cross(&edges[8]))); // N face
+        face_normals.push(Unit::new_normalize(edges[1].cross(&edges[9]))); // E face
+        face_normals.push(Unit::new_normalize(edges[2].cross(&edges[10]))); // S face
+        face_normals.push(Unit::new_normalize(edges[3].cross(&edges[11]))); // W face
+        face_normals.push(Unit::new_normalize(edges[1].cross(&edges[0]))); // down face
+        face_normals.push(Unit::new_normalize(edges[5].cross(&edges[4]))); // up face
```

Comment:
As above, how about using `ArrayVec::from`?

Reply:
You're right, here the length matches the capacity. Done.